src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -79,7 +80,7 @@ private void BuildTraceJsonSerializer()
             _traceSerializer = Newtonsoft.Json.JsonSerializer.Create(jsonSettings);
         }
 
-        private (JsonRpcRequest Model, List<JsonRpcRequest> Collection) DeserializeObjectOrArray(string json)
+        private (JsonRpcRequest Model, List<JsonRpcRequest> Collection) DeserializeObjectOrArray1(string json)
```

Comment:
Is it now unused? Remove?

Reply:
Yes, we respond with an array. That was the simplest implementation. 
To make it separate it's much more complicated.

Reply:
Can we crosscheck how other clients do this?

Reply:
I'm pretty sure they have separate responses.

Reply:
> I'm pretty sure they have separate responses.

then I think we cannot do it this way unless we test every single integration library like web3, ethers.js, blockscout, MyCryptoWallet, etc

Reply:
> then I think we cannot do it this way unless we test every single integration library like web3, ethers.js, blockscout, MyCryptoWallet, etc

We haven't handle this kind of messages in WS at all till now and nobody complained but I get your point. I'll confirm how Geth does it and if it differs than I'll change the implementation.

Reply:
So Geth handles it differently in Web Sockets and IPC. 
https://github.com/ethereum/go-ethereum/issues/23575
I've implemented it the same way as in Geth's IPC